packages/purpose-process/src/services/validators.ts

```diff
@@ -269,10 +273,8 @@ export const assertRequesterIsAllowedToRetrieveRiskAnalysisDocument = async ({
     return;
   }
 
-  const activeProducerDelegation = await readModelService.getActiveDelegation(
-    eserviceId,
-    delegationKind.delegatedProducer
-  );
+  const activeProducerDelegation =
+    await readModelService.getActiveProducerDelegationByEserviceId(eserviceId);
```

Comment:
If this is the same as the agreement documents, also the consumer delegate shall be allowed to retrieve documents (all 4 tenants should: consumer, producer, delegateProducer, delegateConsumer). See the `assertRequesterCanRetrieveConsumerDocuments` validator in agreement-process